tests/shell/main.c

```diff
@@ -82,6 +82,12 @@ int main(void)
     char line_buf[SHELL_DEFAULT_BUFSIZE];
 
     /* define own shell commands */
+    shell_run_once(shell_commands, line_buf, SHELL_DEFAULT_BUFSIZE);
+
+    puts("shell exited");
+
+    /* Restart the shell after the previous one exits, so that we can test
+     * Ctrl-D exit */
     shell_run(shell_commands, line_buf, SHELL_DEFAULT_BUFSIZE);
```

Comment:
I would run this in a loop instead.

Reply:
Also in native I get an infinite loop, can we do something simililar as for #13675?

Reply:
Two different behaviours are tested here. When we do ctrl-d in `shell_run_once` the execution returns to `main` of the test program. If we do ctrl-d in `shell_run` the shell will be respawn. I added a second test case for ctrl-d and comments to emphasize this behaviour. Do you still get an infinite loop?